File: src/main/java/organizaAI/OrganizaAI/entity/Agendamento.java
package organizaAI.OrganizaAI.entity;

import jakarta.persistence.*;
import lombok.Data;
import organizaAI.OrganizaAI.entity.enums.Status;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "agendamentos")
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nomeCliente;

    // Vínculo opcional com um cliente cadastrado
    @ManyToOne(fetch = FetchType.EAGER, optional = true)
    @JoinColumn(name = "cliente_id", nullable = true)
    private Cliente cliente;

    // Relacionamento com Servico — muitos agendamentos para um serviço
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "servico_id", nullable = false)
    private Servico servico;

    @Column(nullable = false)
    private LocalDateTime dataHora;

    // Hora de término calculada com base na duração do serviço
    @Column(nullable = false)
    private LocalDateTime dataHoraFim;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.PENDENTE;
}

File: frontend/lib/models/agendamento.dart
import 'servico.dart';

class Agendamento {
  final int id;
  final int? clienteId;
  final String nomeCliente;
  final String? telefoneCliente;
  final Servico servico;
  final DateTime dataHora;
  final DateTime dataHoraFim;
  final String status;

  Agendamento({
    required this.id,
    this.clienteId,
    required this.nomeCliente,
    this.telefoneCliente,
    required this.servico,
    required this.dataHora,
    required this.dataHoraFim,
    required this.status,
  });

  factory Agendamento.fromJson(Map<String, dynamic> json) {
    return Agendamento(
      id: json['id'],
      clienteId: json['clienteId'],
      nomeCliente: json['nomeCliente'],
      telefoneCliente: json['telefoneCliente'],
      servico: Servico.fromJson(json['servico']),
      dataHora: DateTime.parse(json['dataHora']),
      dataHoraFim: DateTime.parse(json['dataHoraFim']),
      status: json['status'],
    );
  }
}
